source/common/common/matchers.cc

```diff
@@ -32,7 +32,7 @@ ValueMatcherConstSharedPtr ValueMatcher::create(const envoy::type::matcher::v3::
   case envoy::type::matcher::v3::ValueMatcher::MatchPatternCase::kListMatch:
     return std::make_shared<const ListMatcher>(v.list_match());
   default:
-    NOT_REACHED_GCOVR_EXCL_LINE;
+    throw EnvoyException("Uncaught default");
```

Comment:
IIUC, `NOT_REACHED_GCOVR_EXCL_LINE` was originally a signal to the coverage system not to count that line against us in the line-coverage computation. It was also a signal to humans that was the intent.

At some point our coverage infrastructure changed, and `NOT_REACHED_GCOVR_EXCL_LINE` no longer is recognized as a "this doesn't count" signal to the coverage metrics, but it is still a useful signal (IMO) to humans indicating we don't expect the line to be hit, and someday maybe we'll be able to convince the coverage system, once again, not to count such lines.

I feel like IS_ENVOY_BUG and this `throw` are less clear signals for that. On the flip side, your change adds more details in many cases which might be useful to the human reader.

But I wonder if we could keep some kind of coverage-specific macro indicating the intent, with the detail you've added, even if isn't actually recognized (yet) by any of our coverage tools.

WDYT?

Reply:
I think this was already discussed on slack and this is what we came up with :-)

I think also saying "it's a bug to hit this line of code" is a pretty clear indicator that code should not be hit.  If we ever have coverage tools which allow excluding and don't have that name we can always bulk rename then?